util/fork_func.c

```diff
@@ -0,0 +1,77 @@
+#include <sys/wait.h>
+
+#include <stdint.h>
+#include <unistd.h>
+
+#include "warnp.h"
+
+#include "fork_func.h"
+
+/**
+ * fork_func(func, cookie):
+ * Fork and run ${func} in a new process, with ${cookie} as the sole argument.
+ */
+pid_t
+fork_func(int (* func)(void *), void * cookie)
+{
+	pid_t pid;
+
+	/* Fork */
```

Comment:
This is traditionally written with `switch (pid = fork())`, see `fork(2)` for example.

Reply:
sigh, ok, I'll change it.  I still think it's easier to leverage our normal "do something; did it fail?" idiom.  Then if it did not fail, then figure out what type of success it had.

BTW, although I've seen the traditional `switch` method before, it's not in the freebsd 12.4 man page, which has no `EXAMPLES` section:
https://man.freebsd.org/cgi/man.cgi?query=fork&apropos=0&sektion=2&manpath=FreeBSD+12.4-RELEASE

Evidently that was only added in 13.1:
https://reviews.freebsd.org/D27626